packages/server-wallet/src/protocols/__test__/defund-channel.test.ts

```diff
@@ -0,0 +1,175 @@
+import {defaultTestConfig} from '../..';
+import {createLogger} from '../../logger';
+import {DBDefundChannelObjective, ObjectiveModel} from '../../models/objective';
+import {Store} from '../../wallet/store';
+import {testKnex as knex} from '../../../jest/knex-setup-teardown';
+import {seedAlicesSigningWallet} from '../../db/seeds/1_signing_wallet_seeds';
+import {channel} from '../../models/__test__/fixtures/channel';
+import {Channel} from '../../models/channel';
+import {MockChainService} from '../../chain-service';
+import {WalletResponse} from '../../wallet/wallet-response';
+import {ChannelDefunder} from '../defund-channel';
+import {ChallengeStatus} from '../../models/challenge-status';
+import {stateVars} from '../../wallet/__test__/fixtures/state-vars';
+import {stateWithHashSignedBy} from '../../wallet/__test__/fixtures/states';
+import {alice, bob} from '../../wallet/__test__/fixtures/signing-wallets';
+
+const logger = createLogger(defaultTestConfig());
+const timingMetrics = false;
+
+let store: Store;
+beforeEach(async () => {
+  store = new Store(
+    knex,
+    defaultTestConfig().metricsConfiguration.timingMetrics,
+    defaultTestConfig().skipEvmValidation,
+    '0'
+  );
+
+  await store.dbAdmin().truncateDB();
+  await seedAlicesSigningWallet(knex);
+});
+
+test('it should submit a conclude transaction even if there is an active challenge if there is a conclusion proof', async () => {
+  const chainService = new MockChainService();
+  const channelDefunder = ChannelDefunder.create(store, chainService, logger, timingMetrics);
+
+  const c = channel({
+    channelNonce: 1,
+    vars: [stateWithHashSignedBy([alice(), bob()])({isFinal: true})],
+  });
+
+  await Channel.query(knex).withGraphFetched('signingWallet').insert(c);
+
+  const challengeState = {
+    ...c.channelConstants,
+    ...stateVars(),
+  };
+  await ChallengeStatus.insertChallengeStatus(knex, c.channelId, 100, challengeState);
+  const obj = createPendingObjective(c.channelId);
+  await knex.transaction(tx => ObjectiveModel.insert(obj, tx));
+  const response = WalletResponse.initialize();
+  const concludeSpy = jest.spyOn(chainService, 'concludeAndWithdraw');
+
+  await channelDefunder.crank(obj, response);
+
+  const reloadedObjective = await store.getObjective(obj.objectiveId);
+  expect(reloadedObjective.status).toEqual('succeeded');
+  expect(concludeSpy).toHaveBeenCalled();
+});
+
+test('when the channel does not exist it should fail the objective', async () => {
+  const chainService = new MockChainService();
+  const channelDefunder = ChannelDefunder.create(store, chainService, logger, timingMetrics);
+  const c = channel();
+
+  const obj = createPendingObjective(c.channelId);
+  await knex.transaction(tx => ObjectiveModel.insert(obj, tx));
```

Comment:
Why do you need to start a transaction here? Can you abuse javascript and just do something like
```suggestion
  await ObjectiveModel.insert(obj, undefined as any);
```

Reply:
Wouldn't that fail horribly? We use the `tx` that's passed in?

Reply:
I'm not sure if the transaction _javascript_ methods are used internally. My understanding is that the main point of the transaction that gets passed around is to ensure consistency.

I remember now that `ObjectiveModel` does need to be passed a DB connection. So maybe this will do?
```suggestion
  await ObjectiveModel.insert(obj, knex as any);
```